src/qibo/quantum_info/quantum_networks.py

```diff
@@ -629,53 +569,500 @@ def _run_checks(self, partition, system_output, pure):
-            self.system_output = [
-                True,
-            ] * len(self.partition)
-            for k in range(len(self.partition) // 2):
-                self.system_output[k * 2] = False
-            self.system_output = tuple(self.system_output)
+    def full(self, backend=None, update=False):
+        """Convert the internal representation to the full tensor of the network.
+
+        Returns:
+            ndarray: The full reprentation of the quantum network.
+        """
+        if backend is None:  # pragma: no cover
+            backend = self._backend
+        tensor = np.copy(self._tensor)
+        tensor = backend.cast(tensor, dtype=self._tensor.dtype)
+        conj = backend.np.conj
+
+        if self.is_pure():
+            """Reshapes input matrix based on purity."""
+            tensor.reshape(self.dims)
+            if self._backend.__class__.__name__ == "PyTorchBackend":
+                tensor = self._tensordot(tensor, conj(tensor), dims=0)
+            else:
+                tensor = self._tensordot(tensor, conj(tensor), axes=0)
+            tensor = self._operator2tensor(tensor, self.partition)
+
+            if update:
+                self._tensor = tensor
+                self._pure = False
+
+        return tensor
+
+
+class QuantumComb(QuantumNetwork):
+    """Stores a Quantum comb, which is a network in which the systems follows a sequential order.
+
+    It is also called the *non-Markovian quantum process* in many literatures.
+    A quantum comb is a quantum network of the form :math:`J[┍i1┑,┕o1┙,┍i2┑,┕o2┙, ...]`,
+    where the process first take an input state from system :math:`i1`,
+    then output a state to system :math:`o1`, and so on.
+    This is a non-Markovian process as the output of the system :math:`o2` may depend on
+    what happened in systems :math:`i1`, and :math:`o1`.
+
+    A quantum channel is a special case of quantum comb, where there are only one input
+    system and one output system.
+
+    Args:
+        tensor (ndarray): the tensor representations of the quantum Comb.
+        partition (List[int] or Tuple[int]): partition of ``matrix``.
+        system_input (List[bool] or Tuple[bool], optional): mask on the input system of the
+            Choi operator. If ``None``, defaults to
+            ``(True,False,True,False,...)``, where ``len(system_input)=len(partition)``.
+            Defaults to ``None``.
+        pure (bool, optional): ``True`` when ``tensor`` is a "pure" representation (e.g. a pure
+            state, a unitary operator, etc.), ``False`` otherwise. Defaults to ``False``.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): Backend to be used in
+            calculations. If ``None``, defaults to :class:`qibo.backends.GlobalBackend`.
+            Defaults to ``None``.
+    """
+
+    def __init__(
+        self,
+        tensor,
+        partition: Optional[Union[List[int], Tuple[int]]] = None,
+        *,
```

Comment:
It's best to handle this with either `*args` or `**kwargs` if it is really needed.